lighthouse-core/gather/connections/connection.js

```diff
@@ -52,17 +52,21 @@ class Connection {
    * Call protocol methods
    * @template {keyof LH.CrdpCommands} C
    * @param {C} method
+   * @param {string=} sessionId
```

Comment:
I had thought you couldn't have an optional param in the middle but I suppose if the end is rest params... 🤔 

should we just make this thing an object at this point?

Reply:
I really don't want to change the interface to this method.

Reply:
Alternatively, could make another `sendCommandToSession` on Connection

Reply:
meh, probably not worth the trouble and the parallelism with driver at the moment is very nice so IMO this is fine. it was really just a meta point on `string=` vs. `string|undefined` anyhow

Reply:
> it was really just a meta point on `string=` vs. `string|undefined` anyhow

Seems like it should definitely be `string|undefined` to reduce human confusion, and I don't think tsc will let you treat it as optional anyways as long as `LH.CrdpCommands[keyof LH.CrdpCommands]['paramsType']` isn't assignable to `string|undefined`.

Reply:
OK, changed to `string|undefined`.

I agree with the human confusion argument, but FYI `string=` is exactly the same as `string|undefined` from what I can tell. At least, the latter worked fine here and didn't produce TS errors.

Reply:
ah I see it now. It just happened that all the usages of `this.sendCommandToSession` worked the same for either case.